# Indonesia Stock Research Universe

> **Educational research universe—not a buy list.** Inclusion only means an instrument is useful to study. Before investing, verify current availability, legal documents, price/NAV, fees, liquidity, tax, risks, and whether it fits your goals and loss capacity. Names and classifications can change. Starter lists last reviewed: **2026-09-18**.

This notebook has **two layers**: (1) the complete official KSEI registered-share snapshot, and (2) a deliberately small 16-company learning universe. The registry is not a buy list and is not identical to the number of active IDX-listed companies.

In [1]:
from pathlib import Path
import json
import pandas as pd

def locate(relative_path):
    candidates = [Path(relative_path), Path('products/stocks') / relative_path]
    for candidate in candidates:
        if candidate.exists():
            return candidate
    raise FileNotFoundError(f'Could not locate {relative_path}; tried {candidates}')

master_path = locate('data/processed/ksei_registered_share_securities_2026-09-22.csv')
manifest_path = locate('data/manifests/ksei_stock_universe_2026-09-22.json')
registered_securities = pd.read_csv(master_path)
manifest = json.loads(manifest_path.read_text())

summary = pd.Series({
    'KSEI registered share securities': len(registered_securities),
    'Standard four-letter tickers in registry': int(registered_securities['has_standard_four_letter_ticker'].sum()),
    'IDX listed companies (official reference period)': manifest['idx_listed_company_reference_count'],
    'Registry minus listed-company reference': len(registered_securities) - manifest['idx_listed_company_reference_count'],
    'Registry snapshot date': manifest['snapshot_date'],
    'IDX reference period': manifest['idx_listed_company_reference_period'],
}, name='value').to_frame()
summary

,value
KSEI registered share securities,984
Standard four-letter tickers in registry,980
IDX listed companies (official reference period),962
Registry minus listed-company reference,22
Registry snapshot date,2026-09-22
IDX reference period,2026-07


## Layer 1 — Complete registered-share master

The 984 rows below are the complete KSEI **registered share securities** snapshot retrieved on 2026-09-22. This is broader than active IDX-listed ordinary companies: it can include preferred or special classes and registrations whose current status must be checked on the linked KSEI detail page. KSEI's count and IDX's company count therefore must not be substituted for each other.

In [2]:
registered_securities[['ticker', 'registry_name', 'registrar', 'registry_nominal', 'detail_url']].head(25)

,ticker,registry_name,registrar,registry_nominal,detail_url
0,AADI,ADARO ANDALAN INDONESIA Tbk,"DAEN1 - DATINDO ENTRYCOM, PT",3125,https://web.ksei.co.id/services/registered-sec...
1,AALI,ASTRA AGRO LESTARI Tbk,"RISR1 - RAYA SAHAM REGISTRA, PT",500,https://web.ksei.co.id/services/registered-sec...
2,ABBA,MAHAKA MEDIA Tbk,"ADTR1 - ADIMITRA JASA KORPORA, PT",100,https://web.ksei.co.id/services/registered-sec...
3,ABDA,ASURANSI BINA DANA ARTA Tbk,"RISR1 - RAYA SAHAM REGISTRA, PT",500,https://web.ksei.co.id/services/registered-sec...
4,ABMM,ABM INVESTAMA Tbk,"DAEN1 - DATINDO ENTRYCOM, PT",500,https://web.ksei.co.id/services/registered-sec...
5,ACES,ASPIRASI HIDUP INDONESIA Tbk,"ADTR1 - ADIMITRA JASA KORPORA, PT",10,https://web.ksei.co.id/services/registered-sec...
6,ACRO,SAMCRO HYOSUNG ADILESTARI Tbk,"ADTR1 - ADIMITRA JASA KORPORA, PT",20,https://web.ksei.co.id/services/registered-sec...
7,ACST,ACSET INDONUSA Tbk,"SIGU1 - SINARTAMA GUNITA, PT",100,https://web.ksei.co.id/services/registered-sec...
8,ADCP,ADHI COMMUTER PROPERTI Tbk,"DAEN1 - DATINDO ENTRYCOM, PT",100,https://web.ksei.co.id/services/registered-sec...
9,ADES,AKASHA WIRA INTERNATIONAL Tbk,"RISR1 - RAYA SAHAM REGISTRA, PT",1000,https://web.ksei.co.id/services/registered-sec...


In [3]:
query = 'BANK'
registered_securities[
    registered_securities['registry_name'].str.contains(query, case=False, na=False)
][['ticker', 'registry_name', 'detail_url']].head(30)

,ticker,registry_name,detail_url
18,AGRO,BANK RAYA INDONESIA Tbk,https://web.ksei.co.id/services/registered-sec...
19,AGRS,BANK IBK INDONESIA Tbk,https://web.ksei.co.id/services/registered-sec...
33,AMAR,BANK AMAR INDONESIA Tbk,https://web.ksei.co.id/services/registered-sec...
58,ARTO,BANK JAGO Tbk,https://web.ksei.co.id/services/registered-sec...
83,BABP,BANK MNC INTERNASIONAL Tbk,https://web.ksei.co.id/services/registered-sec...
85,BACA,BANK CAPITAL INDONESIA Tbk,https://web.ksei.co.id/services/registered-sec...
90,BANK,BANK ALADIN SYARIAH Tbk,https://web.ksei.co.id/services/registered-sec...
97,BBCA,BANK CENTRAL ASIA Tbk,https://web.ksei.co.id/services/registered-sec...
98,BBHI,ALLO BANK INDONESIA Tbk,https://web.ksei.co.id/services/registered-sec...
99,BBKP,BANK KB INDONESIA Tbk,https://web.ksei.co.id/services/registered-sec...


## Layer 2 — Curated learning universe

These 16 recognizable companies were chosen to teach different business models and sector-specific metrics. They cover only a small fraction of the market. Large or famous does not mean cheap, safe, liquid, or suitable.

In [4]:
import pandas as pd

curated_learning_universe = pd.DataFrame([
  {
    "ticker": "BBCA",
    "company": "PT Bank Central Asia Tbk",
    "sector": "Financials",
    "field": "Private bank",
    "first_metrics": "NIM, CASA, cost of credit, NPL, ROE, valuation"
  },
  {
    "ticker": "BBRI",
    "company": "PT Bank Rakyat Indonesia (Persero) Tbk",
    "sector": "Financials",
    "field": "State-owned bank / microfinance",
    "first_metrics": "NIM, micro credit, CASA, CoC, NPL, ROE"
  },
  {
    "ticker": "BMRI",
    "company": "PT Bank Mandiri (Persero) Tbk",
    "sector": "Financials",
    "field": "State-owned diversified bank",
    "first_metrics": "NIM, CASA, CoC, NPL, ROE, subsidiaries"
  },
  {
    "ticker": "BBNI",
    "company": "PT Bank Negara Indonesia (Persero) Tbk",
    "sector": "Financials",
    "field": "State-owned corporate bank",
    "first_metrics": "Loan mix, NIM, CASA, CoC, NPL, ROE"
  },
  {
    "ticker": "TLKM",
    "company": "PT Telkom Indonesia (Persero) Tbk",
    "sector": "Infrastructures",
    "field": "Telecommunications",
    "first_metrics": "ARPU, subscribers, data traffic, capex, FCF"
  },
  {
    "ticker": "JSMR",
    "company": "PT Jasa Marga (Persero) Tbk",
    "sector": "Infrastructures",
    "field": "Toll roads",
    "first_metrics": "Traffic, tariff, concession life, debt, interest coverage"
  },
  {
    "ticker": "ICBP",
    "company": "PT Indofood CBP Sukses Makmur Tbk",
    "sector": "Consumer non-cyclicals",
    "field": "Packaged food",
    "first_metrics": "Volume, pricing, gross margin, FX, ROIC"
  },
  {
    "ticker": "INDF",
    "company": "PT Indofood Sukses Makmur Tbk",
    "sector": "Consumer non-cyclicals",
    "field": "Integrated food",
    "first_metrics": "Segment margin, commodity exposure, holding discount, debt"
  },
  {
    "ticker": "AMRT",
    "company": "PT Sumber Alfaria Trijaya Tbk",
    "sector": "Consumer non-cyclicals",
    "field": "Minimarket retail",
    "first_metrics": "Same-store sales, store count, inventory, margin, leases"
  },
  {
    "ticker": "KLBF",
    "company": "PT Kalbe Farma Tbk",
    "sector": "Healthcare",
    "field": "Pharmaceuticals and consumer health",
    "first_metrics": "Product mix, margin, R&D, working capital, ROIC"
  },
  {
    "ticker": "PTBA",
    "company": "PT Bukit Asam Tbk",
    "sector": "Energy",
    "field": "Coal mining",
    "first_metrics": "Volume, realized coal price, cash cost, reserves, capex"
  },
  {
    "ticker": "PGAS",
    "company": "PT Perusahaan Gas Negara Tbk",
    "sector": "Energy",
    "field": "Gas transmission/distribution",
    "first_metrics": "Volume, spread, regulation, infrastructure, debt"
  },
  {
    "ticker": "ANTM",
    "company": "PT Aneka Tambang Tbk",
    "sector": "Basic materials",
    "field": "Diversified mining/metals",
    "first_metrics": "Nickel/gold volume, realized price, cash cost, capex"
  },
  {
    "ticker": "INCO",
    "company": "PT Vale Indonesia Tbk",
    "sector": "Basic materials",
    "field": "Nickel mining",
    "first_metrics": "Production, nickel price, cash cost, projects, ownership"
  },
  {
    "ticker": "ASII",
    "company": "PT Astra International Tbk",
    "sector": "Industrials",
    "field": "Diversified conglomerate",
    "first_metrics": "Segment earnings, auto share, finance subsidiaries, NAV"
  },
  {
    "ticker": "GOTO",
    "company": "PT GoTo Gojek Tokopedia Tbk",
    "sector": "Technology",
    "field": "Digital ecosystem",
    "first_metrics": "GTV, contribution margin, cash burn, dilution, path to profit"
  }
])
curated_learning_universe

,ticker,company,sector,field,first_metrics
0,BBCA,PT Bank Central Asia Tbk,Financials,Private bank,"NIM, CASA, cost of credit, NPL, ROE, valuation"
1,BBRI,PT Bank Rakyat Indonesia (Persero) Tbk,Financials,State-owned bank / microfinance,"NIM, micro credit, CASA, CoC, NPL, ROE"
2,BMRI,PT Bank Mandiri (Persero) Tbk,Financials,State-owned diversified bank,"NIM, CASA, CoC, NPL, ROE, subsidiaries"
3,BBNI,PT Bank Negara Indonesia (Persero) Tbk,Financials,State-owned corporate bank,"Loan mix, NIM, CASA, CoC, NPL, ROE"
4,TLKM,PT Telkom Indonesia (Persero) Tbk,Infrastructures,Telecommunications,"ARPU, subscribers, data traffic, capex, FCF"
5,JSMR,PT Jasa Marga (Persero) Tbk,Infrastructures,Toll roads,"Traffic, tariff, concession life, debt, intere..."
6,ICBP,PT Indofood CBP Sukses Makmur Tbk,Consumer non-cyclicals,Packaged food,"Volume, pricing, gross margin, FX, ROIC"
7,INDF,PT Indofood Sukses Makmur Tbk,Consumer non-cyclicals,Integrated food,"Segment margin, commodity exposure, holding di..."
8,AMRT,PT Sumber Alfaria Trijaya Tbk,Consumer non-cyclicals,Minimarket retail,"Same-store sales, store count, inventory, marg..."
9,KLBF,PT Kalbe Farma Tbk,Healthcare,Pharmaceuticals and consumer health,"Product mix, margin, R&D, working capital, ROIC"


In [5]:
official_company_reference = manifest['idx_listed_company_reference_count']
missing_from_registry = sorted(set(curated_learning_universe['ticker']) - set(registered_securities['ticker']))
assert not missing_from_registry, f'Curated tickers missing from KSEI registry: {missing_from_registry}'

pd.Series({
    'curated_companies': len(curated_learning_universe),
    'official_IDX_company_reference': official_company_reference,
    'curated_coverage_percent': round(100 * len(curated_learning_universe) / official_company_reference, 2),
    'IDX_IC_sectors_represented': curated_learning_universe['sector'].nunique(),
}, name='value').to_frame()

,value
curated_companies,16.00
official_IDX_company_reference,962.00
curated_coverage_percent,1.66
IDX_IC_sectors_represented,8.00


In [6]:
curated_learning_universe.groupby(['sector', 'field']).size().rename('companies').to_frame()

companies
sector                 field                                         
Basic materials        Diversified mining/metals                    1
                       Nickel mining                                1
Consumer non-cyclicals Integrated food                              1
                       Minimarket retail                            1
                       Packaged food                                1
Energy                 Coal mining                                  1
                       Gas transmission/distribution                1
Financials             Private bank                                 1
                       State-owned bank / microfinance              1
                       State-owned corporate bank                   1
                       State-owned diversified bank                 1
Healthcare             Pharmaceuticals and consumer health          1
Industrials            Diversified conglomerate                     1
Infrastructures        Telecommunications                           1
                       Toll roads                                   1
Technology             Digital ecosystem                            1

## What is still absent from the curated layer

IDX-IC has 11 broad sectors. The curated sample currently represents 8 and omits **Consumer Cyclicals**, **Properties & Real Estate**, and **Transportation & Logistic**. Future additions should be chosen for learning value and data quality, not merely to make the list longer.

## How to compare banks properly

For BBCA, BBRI, BMRI, and BBNI, build a point-in-time panel from audited reports and official disclosures. Compare loan growth, CASA ratio, NIM, cost-to-income, gross/net NPL, loan-at-risk, cost of credit, CET1/CAR, ROA, ROE, book value per share, P/B, and dividend payout. Align periods and definitions; never mix standalone and consolidated figures silently.

In [7]:
bank_tickers = ['BBCA', 'BBRI', 'BMRI', 'BBNI']
banks = curated_learning_universe[curated_learning_universe['ticker'].isin(bank_tickers)].copy()
banks[['ticker', 'company', 'field', 'first_metrics']]

,ticker,company,field,first_metrics
0,BBCA,PT Bank Central Asia Tbk,Private bank,"NIM, CASA, cost of credit, NPL, ROE, valuation"
1,BBRI,PT Bank Rakyat Indonesia (Persero) Tbk,State-owned bank / microfinance,"NIM, micro credit, CASA, CoC, NPL, ROE"
2,BMRI,PT Bank Mandiri (Persero) Tbk,State-owned diversified bank,"NIM, CASA, CoC, NPL, ROE, subsidiaries"
3,BBNI,PT Bank Negara Indonesia (Persero) Tbk,State-owned corporate bank,"Loan mix, NIM, CASA, CoC, NPL, ROE"


## Primary sources, scope, and data route

- [KSEI registered-shares list](https://web.ksei.co.id/services/registered-securities/shares?setLocale=en-US) supplies the complete 984-security snapshot and one detail URL per code.
- [IDX exchange-activities table](https://www2.idx.co.id/en/market-data/statistical-reports/digital-statistic/monthly/idx-trading-activities/indonesia-stock-exchange-activities?filter=eyJ5ZWFyIjoiMjAyNiIsIm1vbnRoIjoiNyIsInF1YXJ0ZXIiOjAsInR5cGUiOiJtb250aGx5In0%3D) reports 962 listed companies for July 2026.
- [IDX stock-price table](https://www.idx.co.id/id/data-pasar/laporan-statistik/digital-statistic/monthly/trading-summary/table-of-stock-price) is the desired official source for current IDX-IC sector/subsector and market data, but its automated endpoint returned 403 during this snapshot.
- [IDX Data Services](https://data.idx.co.id/) provides official licensed real-time, delayed, end-of-day, and historical products.
- Issuer reports, [OJK data](https://data.ojk.go.id/SJKPublic/Dataset/Dataset), and [BI SEKI](https://www.bi.go.id/en/statistik/ekonomi-keuangan/seki/Default.aspx) support company, banking, and macro analysis.

The master CSV intentionally preserves KSEI terminology. It does **not** invent an IDX-IC sector for all 984 entries. Refresh classifications and current exchange/status before screening or analysis. Store publication date as well as period end to prevent look-ahead bias.